gravitee-apim-plugin/gravitee-apim-plugin-endpoint/gravitee-apim-plugin-endpoint-http-proxy/src/main/java/io/gravitee/plugin/endpoint/http/proxy/HttpProxyEndpointConnector.java

```diff
@@ -68,21 +71,21 @@ public Set<ConnectorMode> supportedModes() {
 
     @Override
     public Completable connect(ExecutionContext ctx) {
-        return Completable.defer(() -> {
-            Request request = ctx.request();
-            ProxyConnector proxyConnector;
-            if (request.isWebSocket()) {
-                proxyConnector = new WebSocketConnector(configuration, httpClientFactory);
-            } else if (isGrpc(request, configuration.getTarget())) {
-                proxyConnector = new GrpcConnector(configuration, grpcHttpClientFactory);
-            } else {
-                proxyConnector = new HttpConnector(configuration, httpClientFactory);
-            }
-            return proxyConnector.connect(ctx);
-        });
+        Request request = ctx.request();
+        return getConnector(request).connect(ctx);
     }
 
-    private static boolean isGrpc(final HttpRequest httpRequest, final String target) {
+    private ProxyConnector getConnector(final HttpRequest request) {
```

Comment:
Isn't it introducing a cpu overhead because of the concurrenthash map ? Do you have a test to show the gain?